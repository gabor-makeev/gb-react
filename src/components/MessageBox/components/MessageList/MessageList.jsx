import style from './MessageList.module.scss';
import List from '@mui/material/List';
import { Message } from '../Message/Message';

export const MessageList = ({ messages = [], gap = '10px' }) => {
  const messageListStyle = {
    gap: gap,
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    justifyContent: 'flex-end',
  };

  return (
    <List
      className={style['message_list']}
      sx={messageListStyle}
      data-testid={'messageList'}
    >
      {messages.map((message, idx) => (
        <Message message={message} key={idx} />
      ))}
    </List>
  );
};
