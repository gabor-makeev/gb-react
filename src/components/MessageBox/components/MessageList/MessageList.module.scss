.message_list {
  //Custom styles should go here
}