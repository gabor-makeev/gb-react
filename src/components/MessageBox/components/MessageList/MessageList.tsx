import style from './MessageList.module.scss';
import List from '@mui/material/List';
import { Message } from '../Message/Message';
import { MessageItem } from 'src/default-types';
import { FC } from 'react';

interface MessageListProps {
  messages?: MessageItem[];
  gap?: string;
}

export const MessageList: FC<MessageListProps> = ({
  messages = [],
  gap = '10px',
}) => {
  const messageListStyle = {
    gap: gap,
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    justifyContent: 'flex-end',
  };

  return (
    <List
      className={style['message_list']}
      sx={messageListStyle}
      data-testid={'messageList'}
    >
      {messages.map((message, idx) => (
        <Message message={message} key={idx} />
      ))}
    </List>
  );
};
