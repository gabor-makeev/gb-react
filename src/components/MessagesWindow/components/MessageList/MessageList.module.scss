@import "src/styles/variables/colors";

.message-list__container {
  width: 100%;
  height: 100%;
  overflow-y: hidden;
  display: flex;
  align-items: flex-end;
}

.gap {
  &-small {
    gap: 5px;
  }

  &-medium {
    gap: 10px;
  }

  &-big {
    gap: 15px;
  }
}

.background-color {
  &-primary {
    background-color: transparent;
  }

  &-secondary {
    background-color: $deepAquamarine;
  }
}

