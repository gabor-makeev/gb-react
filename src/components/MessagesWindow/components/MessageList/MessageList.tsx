import { FC, useEffect, useRef } from 'react';
import style from './MessageList.module.scss';
import { MessageItem } from './components/MessageItem/MessageItem';
import { Message } from 'src/default-types';
import classNames from 'classnames';
import { MUIStyledMessageList } from '../../../MUIStyledComponents/MUIStyledMessageList';

interface MessageListProps {
  userEmail: string;
  messages: Message[];
  gapType?: string;
  backgroundColor?: string;
}

export const MessageList: FC<MessageListProps> = ({
  userEmail,
  messages,
  gapType = 'medium',
  backgroundColor = 'primary',
}) => {
  const scrollAnchor = useRef<HTMLDivElement>(null);

  useEffect(() => {
    scrollAnchor.current?.scrollIntoView();
  }, [messages]);

  const classes = classNames(
    style[`gap-${gapType}`],
    style[`background-color-${backgroundColor}`]
  );

  return (
    <div className={style['message-list__container']}>
      <MUIStyledMessageList className={classes} data-testid={'messageList'}>
        {messages.map((message, idx) => (
          <MessageItem message={message} userEmail={userEmail} key={idx} />
        ))}
        <div ref={scrollAnchor}></div>
      </MUIStyledMessageList>
    </div>
  );
};
