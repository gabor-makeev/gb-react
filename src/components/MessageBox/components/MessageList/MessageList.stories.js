import { MessageList } from './MessageList';
import { DUMMY_CONTENT } from '../../../../constants';

export default {
  title: 'MessageList',
  component: MessageList,
};

const Template = (args) => (
  <MessageList messages={DUMMY_CONTENT.messages} {...args} />
);

export const Primary = Template.bind({});

export const WideMessageSpace = Template.bind({});
WideMessageSpace.args = {
  messageListStyle: {
    gap: '30px',
  },
};
