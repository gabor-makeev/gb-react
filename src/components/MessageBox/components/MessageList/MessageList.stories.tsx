import { MessageList } from './MessageList';
import { DUMMY_CONTENT } from '../../../../constants';
import { ComponentMeta, ComponentStory } from '@storybook/react';

export default {
  title: 'MessageList',
  component: MessageList,
} as ComponentMeta<typeof MessageList>;

const Template: ComponentStory<typeof MessageList> = (args) => (
  <MessageList {...args} messages={DUMMY_CONTENT.messages} />
);

export const Primary = Template.bind({});

export const WideMessageSpace = Template.bind({});
WideMessageSpace.args = {
  gap: '30px',
};
