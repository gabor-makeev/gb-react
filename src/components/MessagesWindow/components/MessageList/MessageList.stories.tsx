import { DUMMY_CONTENT } from 'src/constants';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { MessageList } from './MessageList';

export default {
  title: 'MessageList',
  component: MessageList,
} as ComponentMeta<typeof MessageList>;

const Template: ComponentStory<typeof MessageList> = (args) => (
  <MessageList {...args} messages={DUMMY_CONTENT.messages} />
);

export const Primary = Template.bind({});

export const SmallGaps = Template.bind({});
SmallGaps.args = {
  gapType: 'small',
};

export const BigGaps = Template.bind({});
BigGaps.args = {
  gapType: 'big',
};

export const RoundishAndle = Template.bind({});
RoundishAndle.args = {
  angleType: 'roundish',
};

export const SecondaryBackgroundColor = Template.bind({});
SecondaryBackgroundColor.args = {
  backgroundColor: 'secondary',
};
