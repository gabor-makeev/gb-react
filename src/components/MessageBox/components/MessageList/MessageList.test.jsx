import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { MessageList } from './MessageList';
import { AUTHORS, DUMMY_CONTENT } from '../../../../constants';

describe('MessageList', () => {
  it('should render', () => {
    render(<MessageList messages={DUMMY_CONTENT.messages} />);
  });

  it(`should render ${DUMMY_CONTENT.messages.length} messages`, () => {
    render(<MessageList messages={DUMMY_CONTENT.messages} />);

    expect(screen.getAllByText(AUTHORS.user).length).toBe(
      DUMMY_CONTENT.messages.length
    );
  });

  it('should render without messages passed', () => {
    render(<MessageList />);
  });

  it('should render an 1 unordered list', () => {
    render(<MessageList />);

    expect(screen.getAllByTestId('messageList').length).toBe(1);
  });

  it('should render with default style (messageListStyle) applied', () => {
    render(<MessageList />);

    expect(screen.getByTestId('messageList')).toHaveStyle('gap: 10px');
  });

  it('should render with custom styles passed', () => {
    render(<MessageList messageListStyle={{color: 'red'}} />);

    expect(screen.getByTestId('messageList')).toHaveStyle('color: red');
  });

  it('should render with snapshot', () => {
    const { asFragment } = render(
      <MessageList messages={DUMMY_CONTENT.messages} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
