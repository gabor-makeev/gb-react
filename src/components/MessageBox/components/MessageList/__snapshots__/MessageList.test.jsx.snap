// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should render with snapshot 1`] = `
<DocumentFragment>
  <ul
    class="message_list"
    data-testid="messageList"
    style="gap: 10px;"
  >
    <li
      class="message"
      style="background-color: rgb(49, 232, 101);"
    >
      placerat duis ultricies lacus sed turpis tincidunt id aliquet risus
      <span
        class="message__author-sign"
      >
        Gabor
      </span>
    </li>
    <li
      class="message"
      style="background-color: rgb(49, 232, 101);"
    >
      et molestie ac feugiat sed lectus vestibulum mattis ullamcorper velit
      <span
        class="message__author-sign"
      >
        Gabor
      </span>
    </li>
    <li
      class="message"
      style="background-color: rgb(49, 232, 101);"
    >
      felis eget velit aliquet sagittis id consectetur purus ut faucibus
      <span
        class="message__author-sign"
      >
        Gabor
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
